### PR TITLE
Bluetooth: Mesh: Add a note about Bluetooth Mesh in FOTA upgrade

### PR DESCRIPTION
**Pull request by PavelVPV**
The FOTA upgrade can be performed on a mesh device. The device's
composition data may be different on a new image. To handle the upgrade
correctly for the mesh, the device should be unprovisioned first.

Signed-off-by: Pavel Vasilyev <pavel.vasilyev@nordicsemi.no>

### CODE REVIEW

File: doc/nrf/ug_nrf52.rst [hunk 1]
```diff
@@ -232,6 +232,10 @@ To perform a FOTA upgrade, complete the following steps:
       The :file:`app_update.bin` file is the file that must be downloaded to the device.
 
 #. Download the new image to a device.
+
+      .. note::
+         When performing the FOTA upgrade on a Bluetooth Mesh device and the device's composition data changes, unprovision the device before downloading the new image.
```
Comment (mia-ko) [inline doc/nrf/ug_nrf52.rst:237]:
```suggestion
         When performing FOTA upgrade on a Bluetooth mesh device and the device's composition data changes, unprovision the device before downloading the new image.
```

File: doc/nrf/ug_nrf52.rst [hunk 2]
```diff
@@ -232,6 +232,10 @@ To perform a FOTA upgrade, complete the following steps:
       The :file:`app_update.bin` file is the file that must be downloaded to the device.
 
 #. Download the new image to a device.
+
+      .. note::
+         When performing FOTA upgrade on a Bluetooth mesh device and the device's composition data changes, unprovision the device before downloading the new image.
+
```
Comment (omkar3141) [inline doc/nrf/ug_nrf52.rst:238]: - [ ] 
```suggestion
         When performing FOTA upgrade on a Bluetooth mesh device and if the device's composition data is going to change after DFU, unprovision the device before downloading the new image.

```

Comment (PavelVPV) [reply doc/nrf/ug_nrf52.rst]: Added

File: doc/nrf/ug_nrf52.rst [hunk 3]
```diff
@@ -232,6 +232,10 @@ To perform a FOTA upgrade, complete the following steps:
       The :file:`app_update.bin` file is the file that must be downloaded to the device.
 
 #. Download the new image to a device.
+
+      .. note::
+         When performing FOTA upgrade on a Bluetooth mesh device and if the device's composition data is going to change after the firmware upgrade, unprovision the device before downloading the new image.
```
Comment (mia-ko) [inline doc/nrf/ug_nrf52.rst:237]:
```suggestion
         If the device's composition data is going to change after the firmware upgrade when performing FOTA upgrade on a Bluetooth mesh device, unprovision the device before downloading the new image.
```
It feels a bit easier to read to me, just by swapping it around. See if you agree :)

Comment (Balaklaka) [reply doc/nrf/ug_nrf52.rst:237]: Looking at the documentation preview is looks a little bit strange that we start the "Download the new image to a device." with a note and end with a note. The text in the middle disappears somewhat. Could me merge the two notes into 1 note with two sections or something. This is mostly cosmetic.

**Review by omkar3141 — APPROVED** [review]

**Pull request merged**